src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/Strategy.kt

```diff
@@ -101,15 +106,20 @@ abstract class Strategy protected constructor(
  * @return the failure, if detected, null otherwise.
  */
 fun Strategy.runIteration(invocations: Int, verifier: Verifier): LincheckFailure? {
+    var failure: LincheckFailure? = null
+
     for (invocation in 0 until invocations) {
         if (!nextInvocation())
-            return null
+            break
         val result = runInvocation()
-        val failure = verify(result, verifier)
+        restoreStaticMemorySnapshot()
+        failure = verify(result, verifier)
         if (failure != null)
-            return failure
+            break
     }
-    return null
+
+    restoreStaticMemorySnapshot()
```

Comment:
Should be called from `ManagedStrategy`.

Reply:
But do you need additional call to `restoreStaticMemorySnapshot` here, if there is already a call to it in `ManagedStrategy::run` ?